modules/enum_monitoring/main.py

```diff
@@ -1,351 +1,351 @@
+        session.update(pacu_main.database, Config=config_data)
+        print('  {} total Config rules'.format(len(session.Config['Rules'])))
+        summary_data.update({
+            'config': {
+                'rules': len(all_rules),
+                'recorders': len(all_configuration_recorders),
+                'delivery_channels': len(all_delivery_channels),
+                'aggregators': len(all_configuration_aggregators),
+            }
+        })
+
+    if enum_all is True or args.cloud_watch is True:
+        print('CloudWatch Enumeration...')
+        cw_regions = get_regions('monitoring')
+        all_alarms = []
+
+        for region in cw_regions:
+            print('  {}...'.format(region))
+
+            client = pacu_main.get_boto3_client('cloudwatch', region)
+
+            response = client.describe_alarms()
+            alarms = response['MetricAlarms']
+            while 'NextToken' in response:
+                response = client.describe_alarms(
+                    NextToken=response['NextToken']
+                )
+                alarms.extend(response['MetricAlarms'])
+            print('    {} alarms'.format(len(alarms)))
+
+            for alarm in alarms:
+                alarm['Region'] = region
+
+            all_alarms.extend(alarms)
+
+        cw_data = deepcopy(session.CloudWatch)
+        cw_data['Alarms'] = all_alarms
+        session.update(pacu_main.database, CloudWatch=cw_data)
+        print('  {} total CloudWatch alarms'.format(len(session.CloudWatch['Alarms'])))
+        summary_data['alarms'] = len(all_alarms)
+
+    if enum_all is True or args.vpc is True:
+        print('VPC Enumeration...')
+        vpc_regions = get_regions('ec2')
+        all_flow_logs = []
+
+        for region in vpc_regions:
+            print('  Starting region {}...'.format(region))
+
+            client = pacu_main.get_boto3_client('ec2', region)
+
+            response = client.describe_flow_logs(
+                MaxResults=1000
+            )
+            flow_logs = response['FlowLogs']
+            while 'NextToken' in response:
+                response = client.describe_flow_logs(
+                    MaxResults=1000,
+                    NextToken=response['NextToken']
+                )
+                flow_logs.extend(response['FlowLogs'])
+            print('    {} flow logs'.format(len(flow_logs)))
+
+            for flow_log in flow_logs:
+                flow_log['Region'] = region
+
+            all_flow_logs.extend(flow_logs)
+
+        vpc_data = deepcopy(session.VPC)
+        vpc_data['FlowLogs'] = all_flow_logs
+        session.update(pacu_main.database, VPC=vpc_data)
+        print('  {} total VPC flow logs'.format(len(session.VPC['FlowLogs'])))
+        summary_data['flowlogs'] = len(all_flow_logs)
+
+    print('\n{} completed.\n'.format(module_info['name']))
+    return summary_data
+
+
+def summary(data, pacu_main):
+    out = ''
+    if 'ShieldSubscription' in data:

```

Comment:
Same as the other modules, we still should summarize results that found nothing, we only want to hide ones that weren't even enumerated

Reply:
This should be working as intended, the summary_data fields are only updated if the field is actually enumerated.